Src/WindowsServer/WindowsServer.Shared/Implementation/AzureComputeMetadataHeartbeatPropertyProvider.cs

```diff
@@ -0,0 +1,114 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    internal class AzureComputeMetadataHeartbeatPropertyProvider
+    {
+        internal const string HeartbeatPropertyPrefix = "azInst_"; // to ensure no collisions with base heartbeat properties
+
+        /// <summary>
+        /// Expected fields extracted from Azure IMS to add to the heartbeat properties. 
+        /// Set as internal for testing.
+        /// </summary>
+        internal readonly List<string> ExpectedAzureImsFields = new List<string>()
```

Comment:
Consider typing as `IReadOnlyCollection` and using a string array instead of `List<string>`.
i.e.
```C#
internal readonly IReadOnlyCollection<string> ExpectedAzureImsFields = new string[]
{
    "location",
    //etc...
};
```

That makes it clear that it is supposed to be immutable. (Technically, you could still cast it back to a `string[]` and mutate the contents, but you have to go out of your way.)
```

Reply:
Yes, that is the intention here, thanks for calling that out.